client/my-sites/pages/main.jsx

```diff
@@ -103,14 +114,26 @@ module.exports = React.createClass( {
 				</NavItem>
 			);
 		} );
-	},
+	}
 
 	_setWarning( selectedSite ) {
+		const { translate } = this.props;
 		if ( selectedSite && selectedSite.jetpack && ! selectedSite.hasMinimumJetpackVersion ) {
```

Comment:
`_setWarning` is now being passed a site ID rather than a site object in the component's lifecycle methods. However, in here, we expect the data to be a site object with attributes. I expect this warning to not work at all as is. Now, the site ID value gets automatically boxed, otherwise it would have thrown an error, had the value been a primitive:

```
< Number(123456).isJetpack
> undefined

< 123456.isJetpack
> Uncaught SyntaxError: Invalid or unexpected token
```

Since we're also `mapState`ing the actual `site`, the lowest-impact fix is likely to have the lifecycle methods call this `this._setWarning( this.props.site )` instead.